docker-compose-dev.yml

```diff
@@ -0,0 +1,318 @@
+      - ./chris_backend:/opt/app-root/src
+      - .:/opt/ChRIS_ultron_backEnd
+    environment:
+      - DJANGO_SETTINGS_MODULE=config.settings.local
+      - STORAGE_ENV=fslink
+    depends_on:
+      db_migrate:
+        condition: service_completed_successfully
+      queue:
+        condition: service_started
+      pfcon:
+        condition: service_started
+    restart: unless-stopped
+    networks:
+      - local
+  worker_periodic:
+    image: ghcr.io/fnndsc/cube:5.0.0
+    command:
+      - sh
+      - -c
+      - pip install -r /opt/ChRIS_ultron_backEnd/requirements/local.txt && celery -A core worker -c 2 -l info -Q periodic
+    volumes:
+      - chris_files:/var/chris:rw
+      - ./chris_backend:/opt/app-root/src
+      - .:/opt/ChRIS_ultron_backEnd
+    environment:
+      - DJANGO_SETTINGS_MODULE=config.settings.local
+      - STORAGE_ENV=fslink
+    depends_on:
+      db_migrate:
+        condition: service_completed_successfully
+      queue:
+        condition: service_started
+    restart: unless-stopped
+    networks:
+      - local
+  scheduler:
+    image: ghcr.io/fnndsc/cube:5.0.0
+    command:
+      - sh
+      - -c
+      - pip install -r /opt/ChRIS_ultron_backEnd/requirements/local.txt && celery -A core beat -l info --scheduler django_celery_beat.schedulers:DatabaseScheduler
+    volumes:
+      - chris_files:/var/chris:rw
+      - ./chris_backend:/opt/app-root/src
+      - .:/opt/ChRIS_ultron_backEnd
+    environment:
+      - DJANGO_SETTINGS_MODULE=config.settings.local
+      - STORAGE_ENV=fslink
+    depends_on:
+      db_migrate:
+        condition: service_completed_successfully
+      queue:
+        condition: service_started
+    restart: unless-stopped
+    networks:
+      - local
+  chris_dev_db:
+    image: docker.io/library/postgres:16
+    restart: unless-stopped
+    volumes:
+      - db_data:/var/lib/postgresql/data
+    environment:
+      - POSTGRES_DB=chris_dev
+      - POSTGRES_USER=chris
+      - POSTGRES_PASSWORD=Chris1234
+    networks:
+      - local
+    healthcheck:
+      test: ["CMD", "pg_isready"]
+      interval: 2s
+      timeout: 4s
+      retries: 3
+      start_period: 60s
+  queue:
+    image: docker.io/library/rabbitmq:3
+    restart: unless-stopped
+    networks:
+      - local
+
+  pfcon:
+    container_name: pfcon
+    image: ghcr.io/fnndsc/pfcon:5.2.2
+    environment:
+      COMPUTE_SERVICE_URL: http://pman:5010/api/v1/
+      SECRET_KEY: secret
+      PFCON_USER: pfcon
+      PFCON_PASSWORD: pfcon1234
+      PFCON_INNETWORK: "true"
+      STORAGE_ENV: filesystem
+      STOREBASE_MOUNT: /var/local/storeBase
+    ports:
+      - "5005:5005"
+    volumes:
+      - chris_files:/var/local/storeBase
+    networks:
+      local:
+        aliases:
+          - pfcon.host
+      remote:
+    labels:
+      org.chrisproject.role: "pfcon"
+    user: "1001"
+
+  pman:
+    image: ghcr.io/fnndsc/pman:6.0.1
+    container_name: pman
+    environment:
+      CONTAINER_ENV: docker
```

Comment:
Note: this is a breaking change.

`pman` needs to create containers using Docker. There are two available APIs for this:

- Docker API: simpler code and usage
- Swarm API: more complicated

There are different features available depending on whether the Docker API or Swarm API is used. For example, the Docker API supports configurations `CONTAINER_USER` and `SHM_SIZE`, but does not support GPU. The Swarm API does not support `CONTAINER_USER` nor `SHM_SIZE` but Swarm API does work with GPU.

The discrepancy in features is not a technical limitation but rather just laziness.

While I strongly suggest using `CONTAINER_ENV=docker`, @jbernal0019, who is the main contributor of ChRIS_ultron_backEnd, is against `CONTIANER_ENV=docker` and favors `CONTAINER_ENV=swarm`.

More context: https://github.com/fnndsc/pman?tab=readme-ov-file#swarm-vs-docker

Reply:
Thanks for the feedback.

I believe that the goal for `docker-compose-dev.yml` is for development only, not for production.
having `CONTAINER_ENV=docker` should be enough for quick development.

Reply:
pman was updated in https://github.com/FNNDSC/pman/commit/5b6f202c836ba3b4c8e5e35b00ec12eb1329e712 to support GPUs.